core/src/main/java/org/infinispan/commands/functional/AbstractWriteManyCommand.java

```diff
@@ -20,6 +24,7 @@
    long flags;
    DataConversion keyDataConversion;
    DataConversion valueDataConversion;
+   Map<Object, MetaParamsInternalMetadata> internalMetadataMap;
```

Comment:
Is there no way we can trim this somehow... a ConcurrentHashMap for every functional command is a lot of bloat. Why not just 1 like we did with the other metadata?

Reply:
The internal metadata is stored per key. Each key will have a different version.